packages/mobile/src/identity/commentEncryption.ts

```diff
@@ -292,3 +295,39 @@ function* updatePhoneNumberMappings(newIdentityData: IdentityMetadataInTx[]) {
   yield put(updateE164PhoneNumberSalts(e164NumberToSaltUpdates))
   yield put(updateE164PhoneNumberAddresses(e164NumberToAddressUpdates, addressToE164NumberUpdates))
 }
+
+export function* getAuthSignerForAccount(account: string) {
+  // Use the DEK for authentication if the current DEK is registered with this account
+  const registeredEncryptionKey: Buffer | null = yield call(doFetchDataEncryptionKey, account)
+  let authSigner: AuthSigner | undefined
+  if (registeredEncryptionKey) {
+    const encryptionKeyPrivate: string | null = yield select(dataEncryptionKeySelector)
+    if (!encryptionKeyPrivate) {
+      Logger.error(TAG + 'getAuthSignerForAccount', 'Missing comment key, should never happen.')
+    } else {
+      const encryptionKeyPublic = hexToBuffer(compressedPubKey(hexToBuffer(encryptionKeyPrivate)))
+      if (encryptionKeyPublic !== registeredEncryptionKey) {
+        Logger.warn(TAG + 'getAuthSignerForAccount', 'DEK mismatch.')
+      } else {
+        Logger.info(TAG + 'getAuthSignerForAccount', 'Using DEK for authentication')
+        authSigner = {
```

Comment:
Nit: How about just returning here instead of re-assigning the var and then checking it below?